scipyconference/puns.py

```diff
@@ -1,27 +1,97 @@
-from pathlib import Path
+"""
+Core pun generation functionality for scipyconference.
+
+This module contains the implementation of pun generation functions,
+including reading from JSON files and the main create_puns function.
+"""
+
 import random
+from pathlib import Path
 
-from astropy.table import Table
 import numpy as np
 
 
-def create_puns(number=1):
+def _read_puns_from_json():
     """
-    The essence of SciPy are puns.
+    Read puns from the JSONL file.
 
-    Parameters
-    ----------
+    Reads the community-curated puns from the puns.json file in JSONL format.
+    Each line should contain a JSON object with 'pun' and 'github_username' fields.
 
-    number : int
-        Number of puns to generate
+    :return: List of dictionaries containing pun data.
+        Each dictionary has 'pun' and 'github_username' keys.
+        Returns empty list if file is not found or malformed.
+    :rtype: list
     """
+    pun_path = Path(__file__).parent / "puns.json"
+    puns = []
+    with open(pun_path, "r") as f:
+        for line in f:
+            line = line.strip()
+            if line:
+                import json
 
-    # TODO: no need for astropy
-    pun_path = Path(__file__).parent / Path('puns.json')
-    all_puns = sorted(Table.read(pun_path, format='pandas.json')['pun'])
-    if number is np.inf:
-        # TODO: make this to do something really special
-        return '🦀🦀🦀'
-    selected_puns = random.sample(all_puns, number)
+                data = json.loads(line)
+                puns.append(data)
+    return puns
+
+
+def create_puns(number: int):
+    """
+    Create ``number`` of puns for the SciPy conference.
+
+    This function generates puns for the SciPy conference community. It can work
+    in two modes:
+
+    1. Community-curated puns (default): Reads from puns.json file
+    3. Party mode: Creates a celebration with random emojis when number is np.inf
+
+    :param int number: Number of puns to generate. Use np.inf for party mode.
+    :return: None
+    :rtype: None
+    :raises ImportError: If llamabot is required but not installed.
```

Comment:
This is a remnant I think.

Reply:
Good catch, will resolve, just switched room!